airbyte-integrations/connectors/source-square/source_square/components.py

```diff
@@ -0,0 +1,142 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+import logging
+from dataclasses import InitVar, dataclass
+from typing import Any, ClassVar, Iterable, Mapping, MutableMapping, Optional, Union
+
+import pendulum
+from dataclasses_jsonschema import JsonSchemaMixin
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.declarative.auth import DeclarativeOauth2Authenticator
+from airbyte_cdk.sources.declarative.auth.declarative_authenticator import DeclarativeAuthenticator
+from airbyte_cdk.sources.declarative.auth.token import BearerAuthenticator
+from airbyte_cdk.sources.declarative.stream_slicers import StreamSlicer
+from airbyte_cdk.sources.declarative.types import Record, StreamSlice, StreamState
+from airbyte_cdk.sources.streams.core import Stream
+
+
+@dataclass
+class AuthenticatorSquare(DeclarativeAuthenticator, JsonSchemaMixin):
```

Comment:
This is a good candidate to be added to the CDK. There's an issue for this here: https://github.com/airbytehq/airbyte/issues/17681

Would you be open to suggesting to how implement that ticket? (specifically, first let's specify what the YAML looks like) 

If we can agree on the DSL, would love to accept a PR from you into the CDK

Reply:
Described in https://github.com/airbytehq/airbyte/issues/20226